Go/emitter/enhanced.go

```diff
@@ -0,0 +1,35 @@
+package emitter
```

Comment:
It would be great to have some tests with [examples](https://golang.org/pkg/testing/#hdr-Examples) for the EventEmitter, as well as comments for exported types and methods, which can serve as a documentation when used with GoDoc.

Reply:
It's just an example to show students pattern and implementation, we do not need to prepare How.Rogramming.Works repos as projects with tests, docs and CI

Reply:
@tshemsedinov, I don't insist on this, it's just since some other implementations include tests, it makes sense to create some basic tests here as well. It is also a way to show students the way it should be done properly and it may help them to understand how such examples can be tested.